Network/Units/Server/Server.cpp

```diff
@@ -6,3 +6,160 @@
 */
 
 #include "Server.hpp"
+
+namespace ABINetwork
+{
+
+Server::Server(std::size_t numberMaxPlayer)
+{
+    _io_context = std::make_shared<asio::io_context>();
+    if (!_io_context)
+        throw ABIError("Failed to create the server");
+    _tcp_acceptor = std::make_shared<asio::ip::tcp::acceptor>(*_io_context, asio::ip::tcp::endpoint(asio::ip::tcp::v4(), SERVER_TCP_PORT));
+    _udp_socket = std::make_shared<asio::ip::udp::socket>(*_io_context, asio::ip::udp::endpoint(asio::ip::udp::v4(), SERVER_UDP_PORT));
+    if (!_tcp_acceptor || !_udp_socket)
+        throw ABIError("Failed to create the server");
+    _messageId = 0x0000;
+    _numberMaxPlayer = numberMaxPlayer;
+    _currentNumberPlayer = 0;
+    start();
+}
+
+void Server::start()
+{
+    std::cout << "Server running on TCP port " << _tcp_acceptor->local_endpoint().port() << " and UDP port " << _udp_socket->local_endpoint().port() << "..." << std::endl;
+
+    _startAccept();
+    _startReceive();
+    std::thread server_thread([this]() { _io_context->run(); });
+    server_thread.detach();
+}
+
+void Server::_startAccept()
+{
+    auto socket = std::make_shared<asio::ip::tcp::socket>(*_io_context);
+
+    _tcp_acceptor->async_accept(*socket, [this, socket](const asio::error_code& error) {
+        if (!error) {
+            try {
+                uint32_t token = _generateToken();
+                while (_clients.find(token) != _clients.end())
+                    token = _generateToken();
+
+                if (_numberMaxPlayer > _currentNumberPlayer) {
+                    _currentNumberPlayer++;
+                    _clients[token] = asio::ip::udp::endpoint();
+
+                    std::cout << "Client connected with token: " << "0x" << std::hex << std::setw(8) << std::setfill('0') << token << std::dec <<
```

Comment:
Same as before, just check the previous comment

Reply:
Same.